leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/servlet/ClientServlet.java

```diff
@@ -235,12 +239,20 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws Se
                 ContentFormat contentFormat = contentFormatParam != null
                         ? ContentFormat.fromName(contentFormatParam.toUpperCase())
                         : null;
-
-                // create & process request
-                LwM2mNode node = extractLwM2mNode(target, req);
-                WriteRequest request = new WriteRequest(Mode.REPLACE, contentFormat, target, node);
-                WriteResponse cResponse = server.send(registration, request, TIMEOUT);
-                processDeviceResponse(req, resp, cResponse);
+                if(path.length >= 3 && "attributes".equals(path[path.length - 1])){
```

Comment:
we should move the contentFormat stuff in the `else` block as it does not concern write attributes ?

Reply:
you are right, missed that point, I'll update pull request